### PR TITLE
Aggregated config and dataset level for opt-in-out-urls-scan

### PR DESCRIPTION
**Pull request by AndreaFrancis**
Currently we have spawning urls scan numbers but at split level. This PR add config and dataset aggregation level (Only to summarize numbers).

### CODE REVIEW

File: libs/libcommon/src/libcommon/config.py [hunk 1]
```diff
@@ -270,6 +272,16 @@ class ProcessingGraphConfig:
                 "requires": ["split-first-rows-from-streaming"],
                 "job_runner_version": PROCESSING_STEP_SPLIT_OPT_IN_OUT_URLS_SCAN_VERSION,
             },
+            "config-opt-in-out-urls-scan": {
+                "input_type": "config",
+                "requires": ["split-opt-in-out-urls-scan"],
```
Comment (severo) [inline libs/libcommon/src/libcommon/config.py:277]: I think it also requires the list of split names, right?

File: libs/libcommon/src/libcommon/config.py [hunk 2]
```diff
@@ -270,6 +272,16 @@ class ProcessingGraphConfig:
...
+                "job_runner_version": PROCESSING_STEP_CONFIG_OPT_IN_OUT_URLS_SCAN_VERSION,
+            },
+            "dataset-opt-in-out-urls-scan": {
+                "input_type": "dataset",
+                "requires": ["config-opt-in-out-urls-scan"],
```
Comment (severo) [inline libs/libcommon/src/libcommon/config.py:282]: I think that it should also depend on /config-names

File: services/worker/src/worker/config.py [hunk 3]
```diff
@@ -19,9 +19,9 @@
 WORKER_HEARTBEAT_INTERVAL_SECONDS = 60
 WORKER_KILL_LONG_JOB_INTERVAL_SECONDS = 60
 WORKER_KILL_ZOMBIES_INTERVAL_SECONDS = 10 * 60
-WORKER_MAX_DISK_USAGE_PCT = 90
+WORKER_MAX_DISK_USAGE_PCT = 70
```
Comment (severo) [inline services/worker/src/worker/config.py:22]: I think we should change this in another PR

File: services/worker/src/worker/job_runners/config/opt_in_out_urls_scan_from_streaming.py
Comment (severo) [inline services/worker/src/worker/job_runners/config/opt_in_out_urls_scan_from_streaming.py:1]: The tests are missing for the two new job runners

File: services/worker/src/worker/job_runners/dataset/opt_in_out_urls_scan_from_streaming.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    """Raised when the content of the previous step has not the expected format."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "PreviousStepFormatError", cause, False)
+
+
+def compute_opt_in_out_urls_scan_response(dataset: str) -> Tuple[DatasetOptInOutUrlsScanResponse, float]:
+    logging.info(f"get opt-in-out-urls-scan for dataset={dataset}")
+
+    config_names_response = get_previous_step_or_raise(kinds=["/config-names"], dataset=dataset)
```
Comment (severo) [inline services/worker/src/worker/job_runners/dataset/opt_in_out_urls_scan_from_streaming.py:58]: we depend on /config-names, it should be in the list of requires

File: services/worker/src/worker/job_runners/config/opt_in_out_urls_scan_from_streaming.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "PreviousStepFormatError", cause, False)
+
+
+def compute_opt_in_out_urls_scan_response(dataset: str, config: str) -> Tuple[OptInOutUrlsScanResponse, float]:
+    logging.info(f"get config-opt-in-out-urls-scan for dataset={dataset} config={config}")
+
+    split_names_response = get_previous_step_or_raise(
+        kinds=["/split-names-from-streaming"], dataset=dataset, config=config
```
Comment (severo) [inline services/worker/src/worker/job_runners/config/opt_in_out_urls_scan_from_streaming.py:50]: we depend on /split-names-from-streaming, we should add it to the requires of the step

**Review by severo — COMMENTED** [review]
Some details, but it's good

File: libs/libcommon/src/libcommon/config.py (hunk 1 above)
Comment (AndreaFrancis) [reply libs/libcommon/src/libcommon/config.py:278]: Yes I added it here https://github.com/huggingface/datasets-server/blob/config-dataset-spawning-level/libs/libcommon/src/libcommon/config.py#L283

File: libs/libcommon/src/libcommon/config.py (hunk 2 above)
Comment (AndreaFrancis) [reply libs/libcommon/src/libcommon/config.py]: Yes, I added it here https://github.com/huggingface/datasets-server/blob/config-dataset-spawning-level/libs/libcommon/src/libcommon/config.py#L288 thanks!

File: services/worker/src/worker/job_runners/config/opt_in_out_urls_scan_from_streaming.py
Comment (AndreaFrancis) [reply services/worker/src/worker/job_runners/config/opt_in_out_urls_scan_from_streaming.py:1]: I added them https://github.com/huggingface/datasets-server/blob/config-dataset-spawning-level/services/worker/tests/job_runners/dataset/test_opt_in_out_urls_count.py and https://github.com/huggingface/datasets-server/blob/config-dataset-spawning-level/services/worker/tests/job_runners/config/test_opt_in_out_urls_count.py

File: services/worker/src/worker/config.py (hunk 3 above)
Comment (AndreaFrancis) [reply services/worker/src/worker/config.py]: Sorry, it was a mistake, I removed it from the PR

File: services/worker/src/worker/job_runners/config/opt_in_out_urls_scan_from_streaming.py (hunk 5 above)
Comment (AndreaFrancis) [reply services/worker/src/worker/job_runners/config/opt_in_out_urls_scan_from_streaming.py]: Fixed in https://github.com/huggingface/datasets-server/blob/config-dataset-spawning-level/libs/libcommon/src/libcommon/config.py#L283

File: libs/libcommon/src/libcommon/config.py [hunk 6]
```diff
@@ -276,6 +278,16 @@ class ProcessingGraphConfig:
                 "requires": ["split-opt-in-out-urls-scan"],
                 "job_runner_version": PROCESSING_STEP_SPLIT_OPT_IN_OUT_URLS_COUNT_VERSION,
             },
+            "config-opt-in-out-urls-count": {
+                "input_type": "config",
+                "requires": ["/split-names-from-streaming", "split-opt-in-out-urls-scan"],
```
Comment (severo) [inline libs/libcommon/src/libcommon/config.py:283]: maybe we can depend on `split-opt-in-out-urls-count` now?

File: services/worker/src/worker/job_runners/config/opt_in_out_urls_count.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    num_urls = 0
+    num_scanned_rows = 0
+    try:
+        total = 0
+        pending = 0
+        for split_item in content["splits"]:
+            split = split_item["split"]
+            total += 1
+            try:
+                response = get_response(kind="split-opt-in-out-urls-scan", dataset=dataset, config=config, split=split)
```
Comment (severo) [inline services/worker/src/worker/job_runners/config/opt_in_out_urls_count.py:68]: maybe use `split-opt-in-out-urls-count` instead?

**Review by severo — APPROVED** [review]
Two details:
- maybe depend on `split-opt-in-out-urls-count` instead of `split-opt-in-out-urls-scan` to reduce the size of the data fetched from the database.
- fix the flake8 error.

File: services/worker/src/worker/job_runners/config/opt_in_out_urls_count.py (hunk 7 above)
Comment (AndreaFrancis) [reply services/worker/src/worker/job_runners/config/opt_in_out_urls_count.py]: Yes sure, I forgot to change this one but it was fixed on https://github.com/huggingface/datasets-server/pull/1108/commits/f142efba4f815c1af9b654eb3bdb346dd501b9a7

**Pull request merged**